dlt/common/destination/reference.py

```diff
@@ -657,8 +659,8 @@ def __exit__(
 
 class WithStateSync(ABC):
     @abstractmethod
-    def get_stored_schema(self) -> Optional[StorageSchemaInfo]:
-        """Retrieves newest schema from destination storage"""
+    def get_stored_schema(self, any_schema_name: bool = False) -> Optional[StorageSchemaInfo]:
```

Comment:
I'd rather add a new method but it really do not fit here. this interface assumes that there's a known name of a schema.

my take would be to change signature to 
```
get_stored_schema(self, schema_name: str = None)
```
if None is specified, we load the newest schema, if name is provided we load the newest schema with given name